File: SiliconLabsApp/ViewControllers/AppSelection/SILAppSelectionCollectionViewCell.m
//
//  SILAppSelectionCollectionViewCell.m
//  SiliconLabsApp
//
//  Created by Colden Prime on 1/13/15.
//  Copyright (c) 2015 SiliconLabs. All rights reserved.
//

#import "SILAppSelectionCollectionViewCell.h"
#import "UIColor+SILColors.h"
#import "SILApp.h"
#import "SILBluetoothBrowser+Constants.h"
#import "UIView+SILShadow.h"

@interface SILAppSelectionCollectionViewCell()
@property (weak, nonatomic) IBOutlet UIView *roundedView;

@end

@implementation SILAppSelectionCollectionViewCell

- (void)awakeFromNib {
    [super awakeFromNib];
    [self setupCellAppearence];
}

- (void)setupCellAppearence {
    [self setupIconImageView];
    [self setupCellRoundedAppearance];
}

-(void)layoutSubviews {
    [super layoutSubviews];
    self.layer.masksToBounds = NO;
    self.backgroundColor = UIColor.clearColor;
    [self addShadowWithOffset:SILCellShadowOffset radius:SILCellShadowRadius];
}

- (void)setupIconImageView {
    self.iconImageView.layer.masksToBounds = YES;
}

- (void)setupCellRoundedAppearance {
    self.roundedView.layer.masksToBounds = YES;
    self.roundedView.layer.cornerRadius = CornerRadiusStandardValue;
}

- (void)prepareForReuse {
    [super prepareForReuse];
    _imageView = nil;
    _iconImageView = nil;
    _titleLabel = nil;
    _descriptionLabel = nil;
}

- (void)setFieldsInCell:(SILApp*)appData {
    self.titleLabel.text = appData.title;
    self.descriptionLabel.text = appData.appDescription;
    self.iconImageView.image = [UIImage imageNamed:appData.imageName];
}

@end

File: SiliconLabsApp/ViewControllers/AppSelection/SILAppSelectionCollectionViewCell.swift
//
//  SILAppSelectionCollectionViewCell.swift
//  BlueGecko
//
//  Created by Grzegorz Janosz on 13/05/2021.
//  Copyright © 2021 SiliconLabs. All rights reserved.
//

import Foundation

class SILAppSelectionCollectionViewCell: UICollectionViewCell {
    @IBOutlet private weak var roundedView: UIView?
    
    @IBOutlet weak var imageView: UIView?
    @IBOutlet weak var iconImageView: UIImageView?
    @IBOutlet weak var titleLabel: UILabel?
    @IBOutlet weak var descriptionLabel: UILabel?

    override func awakeFromNib() {
        super.awakeFromNib()
        setupCellAppearence()
    }

    private func setupCellAppearence() {
        setupIconImageView()
        setupCellRoundedAppearance()
    }

    override func layoutSubviews() {
        super.layoutSubviews()
        layer.masksToBounds = false
        backgroundColor = UIColor.clear
        addShadow(withOffset: SILCellShadowOffset, radius: SILCellShadowRadius)
    }

    private func setupIconImageView() {
        iconImageView?.layer.masksToBounds = true
        iconImageView?.backgroundColor = .white
        iconImageView?.tintColor = .sil_regularBlue()
    }

    private func setupCellRoundedAppearance() {
        roundedView?.layer.masksToBounds = true
        roundedView?.layer.cornerRadius = CGFloat(CornerRadiusStandardValue)
    }

    override func prepareForReuse() {
        super.prepareForReuse()
        imageView = nil
        iconImageView = nil
        titleLabel = nil
        descriptionLabel = nil
    }

    func setFieldsIn(_ appData: SILApp?) {
        titleLabel?.text = appData?.title
        titleLabel?.textColor = .sil_regularBlue()
        descriptionLabel?.text = appData?.appDescription
        iconImageView?.image = UIImage(named: appData?.imageName ?? "")
    }
}
